apis/condition_set.go

```diff
@@ -253,7 +257,25 @@ func (r conditionsImpl) MarkTrue(t ConditionType) {
 		Status:   corev1.ConditionTrue,
 		Severity: r.severity(t),
 	})
+	r.markHappy(t)
+}
+
+// MarkTrueWithReason sets the status of t to true with the reason, and then marks the happy condition to
+// true if all other dependents are also true.
+func (r conditionsImpl) MarkTrueWithReason(t ConditionType, reason, messageFormat string, messageA ...interface{}) {
+	// set the specified condition
+	r.SetCondition(Condition{
+		Type:     t,
+		Status:   corev1.ConditionTrue,
+		Reason:   reason,
+		Message:  fmt.Sprintf(messageFormat, messageA...),
+		Severity: r.severity(t),
+	})
+	r.markHappy(t)
+}
 
+// markHappy marks the happy condition to true if all other dependents are also true.
+func (r conditionsImpl) markHappy(t ConditionType) {
```

Comment:
Maybe something like `recomputeHappiness` or similar is more fitting? We're not marking things happy unconditionally, we check recheck if we can and do if we can.

Reply:
I love that method name.